module/os/linux/zfs/zfs_dir.c

```diff
@@ -726,11 +722,17 @@ zfs_rmnode(znode_t *zp)
 		 * which point we'll call zfs_unlinked_drain() to process it).
 		 */
 		dmu_tx_abort(tx);
-		zfs_znode_dmu_fini(zp);
```

Comment:
Are you sure all the callers of zfs_rmnode are ok with removing these 2 zfs_znode_dmu_fini calls?

Reply:
As far as I can tell, the only call site is in `zfs_zinactive`. Right?

This change shifts the `zfs_znode_dmu_fini` calls to `zfs_zinactive`.
Which I believe is where it belongs in the first place, given that it operates on DMU internals.

Actually, I think `zfs_rmnode` belongs into zfs_znode.c as well, but I figured that would just be churn and mess up this diff.